src/scripts/main.js

```diff
@@ -1 +1,84 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
```

Comment:
The first promise will always reject after 3 seconds if no left-click occurs, which is expected. However, if a left-click occurs after 3 seconds, the promise will still reject because the `setTimeout` is not cleared. Consider using `clearTimeout` to prevent the rejection if the promise is resolved by a click.